common/JackRequest.h

```diff
@@ -1647,7 +1647,8 @@ struct JackPropertyChangeNotifyRequest : public JackRequest
     {
         jack_uuid_copy(&fSubject, subject);
         memset(fKey, 0, sizeof(fKey));
-        strncpy(fKey, key, sizeof(fKey)-1);
+        if (key)
+            strncpy(fKey, key, sizeof(fKey)-1);
```

Comment:
now that I look at this... won't it possibly copy more bytes than needed?
if key has length 3 for example, we will copy over some garbage memory.

correct me if I am wrong, but shouldn't the 3rd argument in there instead be something like `min(strlen(key), sizeof(fKey)-1)` ?

Reply:
but that's exactly what strncpy does: the 3rd parameter is the maximum nunber of characters to be copied; strlen(key) gives the number of characters in null-terminated key, which might however be greater than size of the target buffer fKey.